src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,134 @@
 package core.basesyntax;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTORY = .75f;
+    private int size;
+    private int capacity;
+    private Node<K, V>[] dataTable;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        dataTable = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size + 1 > capacity * LOAD_FACTORY) {
+            resize();
+        }
+        if (checkKey(key, value)) {
+            putNewNode(getHash(key), key, value, dataTable);
+            ++size;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> node = dataTable[getIndex(key)];
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public int getCapacity() {
+        return capacity;
+    }
```

Comment:
```suggestion
```
Why do we need it?

Reply:
P.s: you can implement other methods of the Map interface.